Sources/SourceKitBazelBSP/RequestHandlers/SKOptions/CompilerArgumentsProcessor.swift

```diff
@@ -17,112 +17,46 @@
 // specific language governing permissions and limitations
 // under the License.
 
+import BazelProtobufBindings
 import Foundation
 import LanguageServerProtocol
 
 private let logger = makeFileLevelBSPLogger()
 
 enum CompilerArgumentsProcessor {
     // Parses and processes the compilation step for a given target from a larger aquery output.
-    // The parsing step is only necessary because BazelTargetAquerier operates on text. Should become unnecessary once we move to proto.
     static func extractAndProcessCompilerArgs(
-        fromAquery aqueryOutput: String,
+        fromAquery aqueryOutput: Analysis_ActionGraphContainer,
         bazelTarget: String,
         contentToQuery: String,
         language: Language,
         initializedConfig: InitializedServerConfig
     ) -> [String]? {
-        var lines: [String] = aqueryOutput.components(separatedBy: "\n")
-        var idx = -1
-        for (i, line) in lines.enumerated() {
-            var actionPrefix = "action 'Compiling "
-            if language == .swift {
-                actionPrefix += "Swift module \(bazelTarget)'"
-            } else {
-                actionPrefix += "\(contentToQuery)'"
-            }
-            if line.hasPrefix(actionPrefix) {
-                if language == .swift {
-                    if lines[i + 1] != "  Mnemonic: SwiftCompile" { continue }
-                } else {
-                    if lines[i + 1] != "  Mnemonic: ObjcCompile" { continue }
-                    if lines[i + 2] != "  Target: \(bazelTarget)" { continue }
-                }
-                idx = i
-                break
-            }
-        }
-        if idx == -1 {
-            logger.error("No module entry found in the aquery for \(contentToQuery)")
-            return nil
-        }
-        // now, get the first index of the line that starts with "Command Line: ("
-        lines = Array(lines.dropFirst(idx + 1))
-        idx = -1
-        for (i, line) in lines.enumerated() {
-            if line.starts(with: "  Command Line: (") {
-                idx = i
-                // Also skip the swiftc line for swift targets
-                if lines[idx + 1].contains("swiftc") { idx += 1 }
-                break
-            }
-        }
-        if idx == -1 {
-            logger.error("No command line entry found")
+        guard let target = aqueryOutput.targets.first(where: { $0.label == bazelTarget }) else {
+            logger.debug("\(CompilerArgError.targetNotFound(bazelTarget)), privacy: .public")
```

Comment:
Although this looks better than the raw strings, OSLog would actually lose its benefits in this case. So I think it might be better to have the errors directly here, like:

`logger.debug("Target: \(target) not found.")`